server/serverimpl.go

```diff
@@ -151,7 +155,7 @@ func (s *server) Addr() net.Addr {
 	return s.addr
 }
 
-func (s *server) httpHandler(w http.ResponseWriter, req *http.Request) {
+func (s *server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
```

Comment:
Instead of having an exported `ServeHTTP` function in the `server` struct can we make this unexported and wrap it with a struct that has an exported `ServeHTTP` function and use that on line 86 instead?

I don't want to unnecessarily export a function that we don't intend to be called by our users.

Reply:
I wasn't too worried about making the change because `server` itself is already unexported and `ServerHTTP` isn't part of the `OpAMPServer interface` . I did go ahead and make that wrapper though. Let me know if this more-or-less is what you were looking for!